src/app/redux/reducers/ethrdid.ts

```diff
@@ -0,0 +1,55 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { DIDDocument } from 'did-resolver'
+
+export interface EtherdidState {
+  owner: string | null,
+  resolve: DIDDocument
+}
+
+interface ChangeOwnerPayload {
+  owner: string
+}
+
+interface ResolveDidPayload {
+  data: DIDDocument
+}
+
+interface AddDelegatePayload {
+  delegate: string
+}
+
+export const initialState: EtherdidState = {
+  owner: null,
+  resolve: {
+    '@context': 'https://w3id.org/did/v1',
+    id: '',
+    publicKey: []
+  }
+}
+
+const ethrDidSlice = createSlice({
+  name: 'ethrdid',
+  initialState,
+  reducers: {
+    changeOwner (state: EtherdidState, { payload: { owner } }: PayloadAction<ChangeOwnerPayload>) {
+      state.owner = owner
+    },
+    resolveDid (state: EtherdidState, { payload: { data } }: PayloadAction<ResolveDidPayload>) {
+      state.resolve = data
+    },
+    addDelegate (state: EtherdidState, { payload: { delegate } }: PayloadAction<AddDelegatePayload>) {
```

Comment:
Should we store delegates that were not fetch from the document in different variable? Anyway, I would re-fetch the did document after this kind of operations are confirmed